nautobot/extras/tests/test_datasources.py

```diff
@@ -662,3 +685,87 @@ def populate_repo(path, url):
+                MockGitRepo.side_effect = self.populate_repo
+                MockGitRepo.return_value.checkout.return_value = self.COMMIT_HEXSHA
+                MockGitRepo.return_value.diff_remote.return_value = []
+
+                self.repo.set_dryrun(init_repo=True)
+
+                git_repository_diff_origin_and_local(
+                    self.repo.pk, self.dummy_request, self.job_result.pk, dryrun_mode="create"
+                )
+                self.job_result.refresh_from_db()
+
+                self.assertEqual(self.job_result.status, JobResultStatusChoices.STATUS_COMPLETED, self.job_result.data)
+
+                config_context = ConfigContext.objects.filter(
+                    name="Frobozz 1000 NTP servers",
+                    owner_object_id=self.repo.pk,
+                    owner_content_type=ContentType.objects.get_for_model(GitRepository),
+                ).first()
+
+                self.assertIsNone(config_context)
+
+                export_template_device = ExportTemplate.objects.filter(
+                    owner_object_id=self.repo.pk,
+                    owner_content_type=ContentType.objects.get_for_model(GitRepository),
+                    content_type=ContentType.objects.get_for_model(Device),
+                    name="template.j2",
+                ).first()
+                self.assertIsNone(export_template_device)
+
+                export_template_html = ExportTemplate.objects.filter(
+                    owner_object_id=self.repo.pk,
+                    owner_content_type=ContentType.objects.get_for_model(GitRepository),
+                    content_type=ContentType.objects.get_for_model(Device),
+                    name="template2.html",
+                ).first()
+                self.assertIsNone(export_template_html)
+
+                export_template_vlan = ExportTemplate.objects.filter(
+                    owner_object_id=self.repo.pk,
+                    owner_content_type=ContentType.objects.get_for_model(GitRepository),
+                    content_type=ContentType.objects.get_for_model(VLAN),
+                    name="template.j2",
+                ).first()
+                self.assertIsNone(export_template_vlan)
+
+                device = Device.objects.get(name=self.device.name)
+                self.assertIsNone(device.local_context_data)
+
+    def test_git_update_and_dry_run(self, MockGitRepo):
+        with tempfile.TemporaryDirectory() as tempdir:
+            with self.settings(GIT_ROOT=tempdir):
+                MockGitRepo.side_effect = self.populate_repo
+                MockGitRepo.return_value.checkout.return_value = self.COMMIT_HEXSHA
+
+                pull_git_repository_and_refresh_data(self.repo.pk, self.dummy_request, self.job_result.pk)
+                self.job_result.refresh_from_db()
+
+                # Now "resync" the repository, but now those files no longer exist in the repository
+                MockGitRepo.side_effect = self.empty_repo
```

Comment:
And similarly, doesn't this result in the local directory being emptied when the MockGitRepo is instantiated?

Reply:
Nope

Reply:
And same idea here - if we're testing a dry-run where the files are deleted from the remote repository, I would expect the local filesystem to be unmodified?

Reply:
Ive updated the test